modules/gdscript/gdscript_warning.cpp

```diff
@@ -194,50 +194,50 @@ String GDScriptWarning::get_name_from_code(Code p_code) {
 	ERR_FAIL_COND_V(p_code < 0 || p_code >= WARNING_MAX, String());
 
 	static const char *names[] = {
-		"UNASSIGNED_VARIABLE",
-		"UNASSIGNED_VARIABLE_OP_ASSIGN",
-		"UNUSED_VARIABLE",
-		"UNUSED_LOCAL_CONSTANT",
-		"UNUSED_PRIVATE_CLASS_VARIABLE",
-		"UNUSED_PARAMETER",
-		"UNUSED_SIGNAL",
-		"SHADOWED_VARIABLE",
-		"SHADOWED_VARIABLE_BASE_CLASS",
-		"SHADOWED_GLOBAL_IDENTIFIER",
-		"UNREACHABLE_CODE",
-		"UNREACHABLE_PATTERN",
-		"STANDALONE_EXPRESSION",
-		"STANDALONE_TERNARY",
-		"INCOMPATIBLE_TERNARY",
-		"UNTYPED_DECLARATION",
-		"INFERRED_DECLARATION",
-		"UNSAFE_PROPERTY_ACCESS",
-		"UNSAFE_METHOD_ACCESS",
-		"UNSAFE_CAST",
-		"UNSAFE_CALL_ARGUMENT",
-		"UNSAFE_VOID_RETURN",
-		"RETURN_VALUE_DISCARDED",
-		"STATIC_CALLED_ON_INSTANCE",
-		"MISSING_TOOL",
-		"REDUNDANT_STATIC_UNLOAD",
-		"REDUNDANT_AWAIT",
-		"ASSERT_ALWAYS_TRUE",
-		"ASSERT_ALWAYS_FALSE",
-		"INTEGER_DIVISION",
-		"NARROWING_CONVERSION",
-		"INT_AS_ENUM_WITHOUT_CAST",
-		"INT_AS_ENUM_WITHOUT_MATCH",
-		"ENUM_VARIABLE_WITHOUT_DEFAULT",
-		"EMPTY_FILE",
-		"DEPRECATED_KEYWORD",
-		"CONFUSABLE_IDENTIFIER",
-		"CONFUSABLE_LOCAL_DECLARATION",
-		"CONFUSABLE_LOCAL_USAGE",
-		"CONFUSABLE_CAPTURE_REASSIGNMENT",
-		"INFERENCE_ON_VARIANT",
-		"NATIVE_METHOD_OVERRIDE",
-		"GET_NODE_DEFAULT_WITHOUT_ONREADY",
-		"ONREADY_WITH_EXPORT",
+		PNAME("UNASSIGNED_VARIABLE"),
+		PNAME("UNASSIGNED_VARIABLE_OP_ASSIGN"),
+		PNAME("UNUSED_VARIABLE"),
+		PNAME("UNUSED_LOCAL_CONSTANT"),
+		PNAME("UNUSED_PRIVATE_CLASS_VARIABLE"),
+		PNAME("UNUSED_PARAMETER"),
+		PNAME("UNUSED_SIGNAL"),
+		PNAME("SHADOWED_VARIABLE"),
+		PNAME("SHADOWED_VARIABLE_BASE_CLASS"),
+		PNAME("SHADOWED_GLOBAL_IDENTIFIER"),
+		PNAME("UNREACHABLE_CODE"),
+		PNAME("UNREACHABLE_PATTERN"),
+		PNAME("STANDALONE_EXPRESSION"),
+		PNAME("STANDALONE_TERNARY"),
+		PNAME("INCOMPATIBLE_TERNARY"),
+		PNAME("UNTYPED_DECLARATION"),
+		PNAME("INFERRED_DECLARATION"),
+		PNAME("UNSAFE_PROPERTY_ACCESS"),
+		PNAME("UNSAFE_METHOD_ACCESS"),
+		PNAME("UNSAFE_CAST"),
+		PNAME("UNSAFE_CALL_ARGUMENT"),
+		PNAME("UNSAFE_VOID_RETURN"),
+		PNAME("RETURN_VALUE_DISCARDED"),
+		PNAME("STATIC_CALLED_ON_INSTANCE"),
+		PNAME("MISSING_TOOL"),
+		PNAME("REDUNDANT_STATIC_UNLOAD"),
+		PNAME("REDUNDANT_AWAIT"),
+		PNAME("ASSERT_ALWAYS_TRUE"),
+		PNAME("ASSERT_ALWAYS_FALSE"),
+		PNAME("INTEGER_DIVISION"),
+		PNAME("NARROWING_CONVERSION"),
+		PNAME("INT_AS_ENUM_WITHOUT_CAST"),
+		PNAME("INT_AS_ENUM_WITHOUT_MATCH"),
+		PNAME("ENUM_VARIABLE_WITHOUT_DEFAULT"),
+		PNAME("EMPTY_FILE"),
+		PNAME("DEPRECATED_KEYWORD"),
+		PNAME("CONFUSABLE_IDENTIFIER"),
+		PNAME("CONFUSABLE_LOCAL_DECLARATION"),
+		PNAME("CONFUSABLE_LOCAL_USAGE"),
+		PNAME("CONFUSABLE_CAPTURE_REASSIGNMENT"),
+		PNAME("INFERENCE_ON_VARIANT"),
+		PNAME("NATIVE_METHOD_OVERRIDE"),
+		PNAME("GET_NODE_DEFAULT_WITHOUT_ONREADY"),
+		PNAME("ONREADY_WITH_EXPORT"),
 #ifndef DISABLE_DEPRECATED
 		"PROPERTY_USED_AS_FUNCTION",
```

Comment:
These are deprecated, so not sure if it's worth touching.

Reply:
If they're still emitted, they should likely be included too.
If they're no-op and just kept for compat in the enum then yeah we don't need to translate them.

Reply:
They don't do anything, but they appear in the settings.

Reply:
Removing a warning code could potentially break someone's script if used in `@warning_ignore`. This is a rather hypothetical case, but still. We decided not to remove deprecated warning codes just in case, at least for now.

But could we try hiding them from Project Settings without removing the warning codes themselves?